File: app/src/main/java/com/michlindev/moviedownloader/api/ApiClient.java
package com.michlindev.moviedownloader.api;


import com.michlindev.moviedownloader.data.Constants;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {

    private final static String BASE_URL = String.format("%s%s", Constants.YTX_DOMAIN,  Constants.API);
    private static Retrofit retrofit = null;

    public static Retrofit getInstance() {

        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }

        return retrofit;
    }
}

File: app/src/main/java/com/michlindev/moviedownloader/api/ApiClient.kt
package com.michlindev.moviedownloader.api

import com.michlindev.moviedownloader.data.Constants
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {

    private const val BASE_URL = "${Constants.YTX_DOMAIN}${Constants.API}"
    val getClient: ApiService
        get() {
            val retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()

            return retrofit.create(ApiService::class.java)
        }
}